distributed/protocol/compression.py

```diff
@@ -211,6 +218,13 @@ def maybe_compress(payload, min_size=1e4, sample_size=1e4, nsamples=5):
 def decompress(header, frames):
     """ Decompress frames according to information in the header """
     return [
-        compressions[c]["decompress"](frame)
-        for c, frame in zip(header["compression"], frames)
+        _decompress_frame(c, frame) for c, frame in zip(header["compression"], frames)
```

Comment:
Rather than have the helper function here I wonder if this could instead be ...

```python
try:
    return [
        compressions[c]["decompress"](frame)
        for c, frame in zip(header["compression"], frames)
    ] 
except KeyError:
    raise ValueError("Invalid compression method: %s" % method)
```

This is a little less fancy, but may be easier for future maintainers to quickly understand without having to track down the `_decompress_frame` method and the `InvalidCompressionMethodError` type.  Hrm, I guess that's hard though because `c` is no longer in scope.  Maybe we go even less fancy and do this in a for loop?  Or perhaps I'm optimizing too much for low-tech.

Reply:
Sure thing -- simplified out the helper method in https://github.com/dask/distributed/pull/3742/commits/78443f547b0d81310559a4898d336e5df3bb1ee7 , in favor of pulling the erroring key from the `KeyError.args`.